lib/ansible/modules/packaging/language/pip.py

```diff
@@ -503,12 +612,28 @@ def main():
         # Automatically apply -e option to extra_args when source is a VCS url. VCS
         # includes those beginning with svn+, git+, hg+ or bzr+
         has_vcs = False
+
         if name:
             for pkg in name:
-                if bool(pkg and re.match(r'(svn|git|hg|bzr)\+', pkg)):
+                if bool(pkg and _is_vcs_url(pkg)):
                     has_vcs = True
                     break
 
+            distributions = _get_distributions_from_names(module, name)
+            # check invalid combination of arguments
+            if version is not None:
+                if len(distributions) > 1:
+                    module.fail_json(
+                        msg="'version' argument is ambiguous when installing multiple package distributions. "
+                            "Please spefify version restrictions next to each distribution in 'name' argument."
+                    )
+                if distributions[0].has_version_specifier:
+                    module.fail_json(
+                        msg="'version' argument conflicts with version specifier provided along with a package name. "
+                            "Please keep a version specifier, but remove the 'version' argument."
```

Comment:
Better: **The** 'version' argument conflicts with any version specifier provided along with a package name. 
Please keep **the** version specifier, but remove the 'version' argument.